src/lib/Components/Home/ArtistRails/ArtistRail.tsx

```diff
@@ -179,13 +176,38 @@ class ArtistRail extends React.Component<Props, State> {
     }
   }
 
+  refreshData = () => {
+    if (this.refs.rail) {
+      if (this.inflightRequest) {
+        this.inflightRequest.dispose()
+      }
+
+      return new Promise((resolve, reject) => {
+        this.inflightRequest = this.props.relay.refetch({ ...this.props.rail, fetchContent: true }, null, error => {
+          if (error) {
+            console.error("ArtistRail.jsx", error.message)
+
+            this.setState({
+              loadFailed: true,
+            })
+
+            reject(error)
+          } else {
+            this.inflightRequest = null
+            resolve()
+          }
+        })
+      })
+    }
+  }
+
```

Comment:
@alloy - this entire block, which is shared with the the other rails (along with associated refresh code), can be lifted up into either a HOC or a render component in a refactor PR.